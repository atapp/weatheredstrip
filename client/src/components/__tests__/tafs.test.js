import React from 'react';
import { render, cleanup } from 'react-testing-library'
import ReactDOM from 'react-dom';
import Tafs from '../tafs.react';
import 'jest-dom/extend-expect';
import testData from './test_data.json'
afterEach(cleanup)

testData.map(airport => {
  it('taf renders properly', () => {
    const { container } = render(<Tafs data={ airport[ 'TAF' ] }/>)
    const tafs = container.firstChild
    expect(tafs).toMatchSnapshot()
  });
})

it('empty taf renders empty', () => {
  const { container } = render(<Tafs data={ [] }/>)
  const tafs = container.firstChild
  expect(tafs).toMatchSnapshot()
})

it('no taf renders empty', () => {
  const { container } = render(<Tafs />)
  const tafs = container.firstChild
  expect(tafs).toMatchSnapshot()
})
