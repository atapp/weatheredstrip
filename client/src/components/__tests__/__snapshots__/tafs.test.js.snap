// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty taf renders empty 1`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`no taf renders empty 1`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`taf renders properly 1`] = `
<div
  class="tafs"
>
  <div
    class="notif-text"
  >
    TAF CYMX 141738Z 1418/1518 24012G22KT P6SM SCT020 TEMPO 1418/1422
     BKN020
     BECMG 1419/1421 22007KT
  </div>
  <div
    class="notif-text"
  >
    FM142200 VRB03KT P6SM FEW030
  </div>
  <div
    class="notif-text"
  >
    FM150600 05005KT 3SM -SN BKN050 OVC100
  </div>
  <div
    class="notif-text"
  >
    FM150900 06007KT 11/2SM -SN VV020
  </div>
  <div
    class="notif-text"
  >
    FM151100 05010KT 1/2SM SN VV005
  </div>
  <div
    class="notif-text"
  >
    RMK NXT FCST BY 150000Z
  </div>
</div>
`;

exports[`taf renders properly 2`] = `
<div
  class="tafs"
>
  <div
    class="notif-text"
  >
    TAF EGPK 141653Z 1418/1518 18009KT 9999 FEW030
  </div>
  <div
    class="notif-text"
  >
    PROB40 TEMPO 1512/1518 19015G25KT
  </div>
</div>
`;

exports[`taf renders properly 3`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`taf renders properly 4`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;
